### PR TITLE
Add tomcat session metrics

### PR DESCRIPTION
**Pull request by checketts**
Addresses #257 

@jkschneider It looks like Tomcat has many metrics available via JMX. Would it make sense to pull those values from JMX and expose as metrics?

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java [hunk 1]
```diff
@@ -263,6 +263,18 @@ public Search value(Statistic statistic, double value) {
             return this;
         }
 
+        public Optional<Double> firstValue(){
```
Comment (jkschneider) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java]: I know you've tried to create this before :). My objection to it is it is only useful for counters and gauges that have a single measurement.

File: micrometer-core/src/test/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        }
+        StandardSession expiredSession = new StandardSession(manager);
+        expiredSession.setId("third");
+        expiredSession.setCreationTime(System.currentTimeMillis()- 10_000);
+        manager.remove(expiredSession, true);
+
+        List<Tag> tags = Tags.zip("metricTag", "val1");
+        TomcatMetrics.monitor(registry, manager, tags);
+
+        assertThat(registry.find("tomcat.sessions.active.max").tags(tags).firstValue()).hasValue(3.0);
```
Comment (jkschneider) [inline micrometer-core/src/test/java/io/micrometer/core/instrument/binder/tomcat/TomcatMetricsTest.java]: Alternative to `firstValue`:
`assertThat(registry.find("tomcat.sessions.active.max").tags(tags).gauge().map(Gauge::value)).isEqualTo(3.0);`

**Comment by checketts** [comment]
@jkschneider Sorry I wasn't trying to sneak that in. I meant to ask for the alternative. I've removed it now.

As you can tell I really prefer getting values in the test to get feedback that way instead of `was not present` errors when using `.value(Statistic.Count, 3)`. In fact if you are on board I would be happy to remove that method and update the unit tests to use the `map()` method.

That being said I'm also not completely in love with the `map` approach since it isn't intuitive/easy to discover, but if all the tests use it, that should improve.

**Comment by jkschneider** [comment]
> It looks like Tomcat has many metrics available via JMX
@checketts Are these in addition to what you have wired through `Manager`?

**Comment by checketts** [comment]
The ones available via JMX are ones that would give it parity with the Jetty metrics. See here: https://www.datadoghq.com/blog/monitor-tomcat-metrics/

As in `tomcat.error_count`, `tomcat.request_count`, `tomcat.max_time`, and `tomcat.processing_time` and more

**Comment by jkschneider** [comment]
Yep, seems like that would be useful to have. No overlap with the current gauges.

**Comment by checketts** [comment]
OK I'll pursue the JMX metrics as a separate PR, if that is OK.

**Pull request merged**